src/tools/manifest2json.ts

```diff
@@ -25,59 +35,65 @@ Usage
   $ tools/sigh manifest2json [options] [file ...]
 
 Description
-  Serializes manifests to JSON. 
+  Serializes manifests to a JSON file. 
 
 Options
+  --outfile, -f output filename; required
   --outdir, -d  output directory; defaults to '.'
-  --outfile, -f output filename; if omitted, generated from the manifest name
   --help        usage info
 `);
   process.exit(0);
 }
 
-if (opts._.some((file) => !file.endsWith('.arcs'))) {
-  console.error(`Only Arcs manifests ('*.arcs') are allowed.`);
+if (!opts.outfile) {
+  console.error(`Parameter --outfile is required.`);
   process.exit(1);
 }
 
-/** Format the name of the output file. */
-function outputName(baseName: string): string {
-  return baseName.replace(/\.arcs$/, '.json');
+if (opts._.some((file) => !file.endsWith('.arcs'))) {
+  console.error(`Only Arcs manifests ('*.arcs') are allowed.`);
+  process.exit(1);
 }
 
 /** Extract JSON serializations from manifest. */
-function toLiteral(manifest: Manifest): object {
-  const lit = {};
-  lit['particles'] = manifest.particles.map(p => p.toLiteral());
-  lit['schemas'] = Object.values(manifest.schemas).map(s => s.toLiteral());
+function toLiteral(manifest: Manifest): Serialization {
+  const lit = new Serialization();
+  lit.particles = manifest.particles.map(p => p.toLiteral());
+  lit.schemas = Object.values(manifest.schemas).map(s => s.toLiteral());
   return lit;
 }
 
-/** Write file serializations for each input file. */
-async function processFile(src: string) {
-  if (!fs.existsSync(src)) {
-    throw new Error(`File not found: ${src}`);
-  }
-
-  const outName = opts.outfile || outputName(path.basename(src));
-  const outPath = path.join(opts.outdir, outName);
+/** Write literals to a file. */
+function processFile(literals: Serialization) {
+  const outPath = path.join(opts.outdir, opts.outfile);
   console.log(outPath);
-
-  const manifest: Manifest = await Runtime.parseFile(src);
-  if (manifest.errors.length) {
-    return;
-  }
-
   const outFile = fs.openSync(outPath, 'w');
-  fs.writeSync(outFile, JSON.stringify(toLiteral(manifest)));
+  fs.writeSync(outFile, JSON.stringify(literals));
   fs.closeSync(outFile);
 }
 
+/** Parse manifests and aggregate literals into a single object. */
+async function aggregateLiterals(srcs: string[]): Promise<Serialization> {
+  const aggregate = new Serialization();
+  for (const src of srcs) {
+    if (!fs.existsSync(src)) {
+      throw new Error(`File not found: ${src}`);
+    }
+    const manifest: Manifest = await Runtime.parseFile(src);
+    if (manifest.errors.length) {
+      throw new Error(`Errors found in manifest '${src}'.`);
```

Comment:
Would he helpful to include the parse errors in the error message here

Reply:
Thanks, that's a good suggestion.